ayon_bin_distro/gui/progress_ui.py

```diff
@@ -51,18 +61,29 @@ def __init__(self, progress, parent):
 
         layout.addWidget(progress_bar)
 
-        self._progress_bar = progress_bar
-        self._icon_label = icon_label
-        self._text_label = text_label
+        self._progress_bar: QtWidgets.QProgressBar = progress_bar
+        self._icon_label: Optional[QtWidgets.QLabel] = icon_label
+        self._text_label: Optional[QtWidgets.QLabel] = text_label
 
     def is_finished(self):
+        """check if the connected progress item has finished executing.
+
+        Returns: bool: true if progress finished
+
+        """
         return self._progress.finished and not self._progress.failed
 
     @property
     def progress_bar(self):
+        """returns the underlying QProgressBar instance
+
+        Returns: QProgressBar instance
+
+        """
         return self._progress_bar
 
     def update_progress(self):
+        """updates the lables and Progress Bars"""
```

Comment:
```suggestion
        """updates the labels and Progress Bars"""
```

By the way, I personally prefer starting the docstrings with uppercase character because it's just a "human-readable sentence" and hence adheres to that grammar. Any reason you start with lowercase?